src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -3620,73 +3612,42 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
 
     void visit_If(const ASR::If_t &x) {
         this->visit_expr_wrapper(x.m_test, true);
-        llvm::Value *cond=tmp;
-        llvm::Function *fn = builder->GetInsertBlock()->getParent();
-        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
-        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
-        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
-        builder->CreateCondBr(cond, thenBB, elseBB);
-        builder->SetInsertPoint(thenBB);
-        for (size_t i=0; i<x.n_body; i++) {
-            this->visit_stmt(*x.m_body[i]);
-        }
-        builder->CreateBr(mergeBB);
-
-        start_new_block(elseBB);
-        for (size_t i=0; i<x.n_orelse; i++) {
-            this->visit_stmt(*x.m_orelse[i]);
-        }
-
-        start_new_block(mergeBB);
+        llvm_utils->create_if_else(tmp, [=]() {
```

Comment:
Using the `llvm_utils->create_if_else` causes `seg fault`. How do I fix this?
@certik @czgdp1807

Reply:
I don't know. You will have to debug. Add `std::cout` statements or use debugger to narrow down to the line which is causing seg fault.

Reply:
Yup, I tried to move `create_if_else` into `llvm_utils`, but it caused `seg fault`.